lib/Compilation.js

```diff
@@ -1326,7 +1320,7 @@ class Compilation extends Tapable {
 			let lengthFreeModules = nextFreeModuleId = usedIdMax + 1;
 
 			while(lengthFreeModules--) {
-				if(!usedIdMap[lengthFreeModules]) {
+				if(!usedIds.has(lengthFreeModules)) {
 					unusedIds.push(lengthFreeModules);
 				}
 			}
```

Comment:
Another thing I wish I could write more functionally but would require an immediately invoked generator...

Or an `Array.from` + `.filter` call, but then we have 2 intermediate arrays.

Reply:
(commenting here because github won't let me comment on the relevant code)

The block after this one does a `for` loop over `this.modules`; can't that also be done as a `for of` on the `modulesIterable`?

Reply:
I decided to go to more imperative style to reduce the amount of created objects, arrays and closures. Using `Array.from` or creating new small function would be counter-productive.

Reply:
> The block after this one does a `for` loop over `this.modules`; can't that also be done as a `for of` on the `modulesIterable`?

It could be. I left it untouched because I was focusing on `Map` and `forEachChunk`.